python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py

```diff
@@ -0,0 +1,223 @@
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class FunctionCallingStepwisePlanner(KernelBaseModel):
+    service_id: str
+    options: FunctionCallingStepwisePlannerOptions
+    generate_plan_yaml: str
+    step_prompt: str
+
+    def __init__(self, service_id: str, options: Optional[FunctionCallingStepwisePlannerOptions] = None):
+        """Initialize a new instance of the FunctionCallingStepwisePlanner
+
+        The FunctionCallingStepwisePlanner is a planner based on top of an OpenAI Chat Completion service
+        (whether it be AzureOpenAI or OpenAI), so that we can use tools.
+
+        If the options are configured to use callbacks to get the initial plan and the step prompt,
+        the planner will use those provided callbacks to get that information. Otherwise it will
+        read from the default yaml plan file and the step prompt file.
+
+        Args:
+            service_id (str): The service id
+            options (Optional[FunctionCallingStepwisePlannerOptions], optional): The options for
+                the function calling stepwise planner. Defaults to None.
+        """
+        options = options or FunctionCallingStepwisePlannerOptions()
+        generate_plan_yaml = (
+            options.get_initial_plan() if options.get_initial_plan else open(PLAN_YAML_FILE_PATH).read()
+        )
+        step_prompt = options.get_step_prompt() if options.get_step_prompt else open(STEP_PROMPT_FILE_PATH).read()
+        options.excluded_plugins.add(STEPWISE_PLANNER_PLUGIN_NAME)
+
+        super().__init__(
+            service_id=service_id,
+            options=options,
+            generate_plan_yaml=generate_plan_yaml,
+            step_prompt=step_prompt,
+        )
+
+    async def invoke(
+        self,
+        kernel: Kernel,
+        question: str,
+    ) -> FunctionCallingStepwisePlannerResult:
+        """
+        Execute the function calling stepwise planner
+
+        Args:
+            kernel: The kernel instance
+            question: The input question
+
+        Returns:
+            FunctionCallingStepwisePlannerResult: The result of the function calling stepwise planner
+
+        Raises:
+            PlannerInvalidConfigurationError: If the input question is empty
+        """
+        if not question:
+            raise PlannerInvalidConfigurationError("Input question cannot be empty")
+
+        try:
+            chat_completion = kernel.get_service(service_id=self.service_id, type=OpenAIChatCompletion)
+        except Exception as exc:
+            raise PlannerInvalidConfigurationError(
+                f"The OpenAI service `{self.service_id}` is not available. Please configure the AI service."
+            ) from exc
+
+        prompt_execution_settings: OpenAIChatPromptExecutionSettings = (
+            self.options.execution_settings
+            or chat_completion.get_prompt_execution_settings_class()(service_id=self.service_id)
```

Comment:
you can use `chat_completion.instantiate_prompt_execution_settings(service_id=service_id)`